lib/merge.js

```diff
@@ -6,9 +6,13 @@ function merge (original, rels) {
     , rel = rels[id]
   if (rel) {
     rels[id] = null
-    var doc = _.defaults.apply(null, [original].concat(rel))
-    doc.sources = _.compact(_.flatten(_.pluck(rel, 'sources')))
-    return [ doc ]
+    if (rel.length) {
+      var doc = _.defaults.apply(null, [original].concat(rel))
+      doc.sources = _.compact(_.flatten(_.pluck(rel, 'sources')))
```

Comment:
I personally dislike the chaining of `loadash`, mostly because it sometimes leads to unexpected behavior. At least it sometimes behaves a little bit different than the chaining of `underscore` that I am used to.